src/utils.c

```diff
@@ -67,3 +71,96 @@ void rv_clock_gettime(struct timespec *tp)
     tp->tv_sec = tv_sec;
     tp->tv_nsec = tv_usec / 1000; /* Transfer to microseconds */
 }
+
+char *sanitize_path(const char *input)
+{
+    size_t n = strnlen(input, MAX_PATH_LEN);
+
+    char *ret = malloc(MAX_PATH_LEN + 1);
+    memset(ret, '\0', MAX_PATH_LEN + 1);
+
+    /* After sanitization, the new path will only be shorter than the original
+     * one. Thus, we can reuse the space */
+    if (n == 0) {
+        ret[0] = '.';
```

Comment:
If `n` equals 0, there is no need to allocate space larger than `MAX_PATH_LEN` for `ret`, correct? I expect a memory-efficient approach.

Reply:
Improved